application-licensing-licensor/application-licensing-licensor-api/src/main/java/com/xwiki/licensing/internal/upgrades/notifications/ExtensionEventConverter.java

```diff
@@ -56,14 +56,7 @@ public class ExtensionEventConverter implements RecordableEventConverter
     @Override
     public Event convert(RecordableEvent recordableEvent, String source, Object data) throws Exception
     {
-        String upgradeMessage = (String) data;
-        Event convertedEvent = this.defaultConverter.convert(recordableEvent, source, data);
-
-        // Before 12.6 or if you use hibernate event store, the parameters you put in an event are not stored so
-        // instead the message is stored directly on the notification body.
-        convertedEvent.setBody(upgradeMessage);
-
-        return convertedEvent;
+        return this.defaultConverter.convert(recordableEvent, source, data);
```

Comment:
I think this is not yet complete, since right now the information is still taken from the body https://github.com/xwikisas/application-licensing/blob/master/application-licensing-licensor/application-licensing-licensor-api/src/main/resources/templates/newVersionAvailable.vm#L26

Reply:
Indeed. There was an error for the extension upgrade notifications but I think I was reproducing in the previous versions as well because there is no code that links a document to the event. Either this, or the `RecordableEvent` class was changed since version 11.10.